packages/ember-metal/lib/property_set.js

```diff
@@ -49,12 +50,16 @@ export function set(obj, keyName, value, tolerant) {
     return setPath(obj, keyName, value, tolerant);
   }
 
+  let possibleDesc = descriptorFor(obj, keyName);
+
+  if (possibleDesc !== undefined) { /* computed property */
+    possibleDesc.set(obj, keyName, value);
+    return value;
+  }
+
   let currentValue = obj[keyName];
```

Comment:
This would also access `obj[keyName]` twice (same problem as `property_get`, see the comment there). However, there were no failing tests for this. Does it matter? Should I add one and change this to use `isDescriptor` instead?